freud/box.pyx

```diff
@@ -33,6 +34,54 @@ logger = logging.getLogger(__name__)
 # _always_ do that, or you will have segfaults
 np.import_array()
 
+
+def _support_1d(func):
+    R"""Decorator for class methods that accept one dimensional input or
+    two dimensional input for the first argument.
+    """
+    @wraps(func)
+    def wrapper(self, arg1, *args, **kwargs):
```

Comment:
Does this decorator work if you specify the input array as a keyword argument?

Reply:
Nope.
```
def _support_1d(func):
    R"""Decorator for class methods that accept one dimensional input or
    two dimensional input for the first argument.
    """
    @wraps(func)
    def wrapper(self, *args, **kwargs):
        if len(args):
            args1 = args[0]
        else:
            args1 = next(iter(kwargs))
        arg1 = np.asarray(arg1)
        if len(arg1.shape) == 1:
            flat = True
            arg1 = np.atleast_2d(arg1)
        else:
            flat = False

        ret = func(self, arg1)
        if flat:
            return ret.squeeze()
        else:
            return ret
    return wrapper
```
is the only way I can think of to handle that case.

Reply:
For `_support_1d_2args`, it would be more complicated and might not worth making a decorator.

Reply:
Seems like the only way to do better would be this method, which would break our Python 2 compatibility (the older Py2-compatible way is considered legacy/deprecated). Even if we didn't care about compatibility, this strikes me as over-engineered.
https://docs.python.org/3/library/inspect.html#inspect.getfullargspec

In the case that we can't support both positional and keyword arguments, I think we have to just write the logic directly (`flat = ...`; `if flat: squeeze...` etc.) into each of the functions... even though it removes the niceness of having a decorator. 😦 It would be silly (and API-breaking) to require either positional or keyword args from a user.